libmscore/scorefile.cpp

```diff
@@ -451,7 +454,18 @@ bool MasterScore::saveFile(bool generateBackup)
 #endif
 #endif
                   }
-            QString backupName = QString(".") + info.fileName() + QString(",");
+            const QString backupName = QString(".") + info.fileName() + QString(",");
+
+            // backup files prior to 3.5 were saved in the same directory as the file itself.
+            // remove these old backup files if needed
+            if (dir != backupDir && dir.exists(backupName)) {
+                  if (!dir.remove(backupName)) {
+//                      if (!MScore::noGui)
+//                            QMessageBox::critical(0, QObject::tr("Save File"),
+//                               tr("Removing old backup file %1 failed").arg(backupName));
+                        }
+                  }
```

Comment:
There's already code for that several lines below. Just modify the conditional to `if (dir != backupDir && dir.exists(backupName))` would be good.

Reply:
done